tests/src/Functional/DeveloperSyncTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\Tests\apigee_edge\Functional;
+
+use Drupal\apigee_edge\Entity\Developer;
+use Drupal\apigee_edge\Entity\DeveloperInterface;
+use Drupal\user\Entity\User;
+use Drupal\user\UserInterface;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Output\BufferedOutput;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Developer synchronization test.
+ *
+ * @group apigee_edge9
+ */
+class DeveloperSyncTest extends ApigeeEdgeFunctionalTestBase {
+
+  public static $modules = [
+    'block',
+  ];
+
+  /**
+   * Random property prefix.
+   *
+   * @var string
+   */
+  protected $prefix;
+
+  /**
+   * Email filter.
+   *
+   * @var string
+   */
+  protected $filter;
+
+  /**
+   * Array of Apigee Edge developers.
+   *
+   * @var \Drupal\apigee_edge\Entity\DeveloperInterface[]
+   */
+  protected $edgeDevelopers = [];
+
+  /**
+   * Array of Drupal users.
+   *
+   * @var \Drupal\user\UserInterface[]
+   */
+  protected $drupalUsers = [];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->prefix = $this->randomMachineName();
+    $escaped_prefix = preg_quote($this->prefix);
+    $this->filter = "/^{$escaped_prefix}\.[a-zA-Z0-9]*@example\.com$/";
+    $this->container->get('config.factory')->getEditable('apigee_edge.sync')->set('filter', $this->filter)->save();
+
+    // Create developers on Apigee Edge.
+    for ($i = 0; $i < 1; $i++) {
+      $this->edgeDevelopers[$i] = Developer::create([
+        'email' => "{$this->prefix}.{$this->randomMachineName()}@example.com",
+        'userName' => $this->randomMachineName(),
+        'firstName' => $this->randomMachineName(),
+        'lastName' => $this->randomMachineName(),
+      ]);
+      $this->edgeDevelopers[$i]->save();
+    }
+
+    // Create users in Drupal.
+    _apigee_edge_set_sync_in_progress(TRUE);
```

Comment:
Add a comment that you are doing this because you do not want to create developer on Apigee Edge.

Reply:
^ 

Reply:
@peterserfozo  ping ^